client/client.go

```diff
@@ -219,6 +226,17 @@ func WithHTTPClient(client *http.Client) func(*Client) error {
 	}
 }
 
+// WithHijackDialer overrides the client default hijack dialer.
+// The typical usecase of WithHijackDialer is to allow implementing connection helpers.
```

Comment:
What is a connection helper? 

Isn't the "typical" use for a HijackDialer to support attach/exec ?